src/sqlfluff/utils/reflow/reindent.py

```diff
@@ -1654,77 +1657,93 @@ def lint_line_length(
                 # We can only fix _inline_ comments in this way. Others should
                 # just be flagged as issues.
                 and line_buffer[-1].segments[-1].is_type("inline_comment")
-                and len(line_buffer[-1].segments[-1].raw) + len(current_indent)
-                <= line_length_limit
             ):
                 reflow_logger.debug("    Handling as inline comment line.")
-                comment_seg = line_buffer[-1].segments[-1]
-                # It is! Move the comment to the line before.
-                fixes = [
-                    # Remove the comment from it's current position, and any whitespace
-                    # in the previous point.
-                    LintFix.delete(comment_seg),
-                    *[
-                        LintFix.delete(ws)
-                        for ws in line_buffer[-2].segments
-                        if ws.is_type("whitespace")
-                    ],
-                ]
-                # Reinsert it at the start of the current line, with a newline after it.
-                if last_indent_idx:
-                    fixes.append(
-                        # NOTE: This looks a little convoluted, but we create *before*
-                        # a block here rather than *after* a point, because the point
-                        # may have been modified already by reflow code and may not be
-                        # a reliable anchor.
-                        LintFix.create_before(
-                            elements[last_indent_idx + 1].segments[0],
-                            [
-                                comment_seg,
-                                NewlineSegment(),
-                                WhitespaceSegment(current_indent),
-                            ],
-                        )
+
+                # Sense check a few edge cases:
+                if "noqa" in line_buffer[-1].segments[-1].raw:
```

Comment:
Approved, but is this an ideal way to check noqa? Wouldn't this also fail on SELECT * FROM myschema.mynoqatable? Would we not want to verify type as a Comment?

Reply:
ah - good thought - but we've just confirmed that it's a comment on line 1659, so I think this is safe here.